src/types/fetch.rs

```diff
@@ -219,6 +220,17 @@ impl<'a> Fetch<'a> {
         })
     }
 
+    /// Extract the `X-GM-LABELS` of a `FETCH` response
+    ///
+    /// This is a Gmail-specific extension. See their
+    /// [developer's page](https://developers.google.com/gmail/imap/imap-extensions) for details.
+    pub fn gmail_labels(&self) -> Option<&Vec<Cow<'a, str>>> {
```

Comment:
I think this will actually end up being nicer to use:
```suggestion
    pub fn gmail_labels(&self) -> Option<impl Iterator<Item = &'a str>> {
```

Reply:
Updated. Pretty new to rust, not sure how idiomatic it is to return a `Map` directly like I'm doing, or if there's a better way entirely.